server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -108,17 +131,20 @@ public class ReadOnlyEngine extends Engine {
      *                   the lock won't be obtained
      * @param readerWrapperFunction allows to wrap the index-reader for this engine.
      * @param requireCompleteHistory indicates whether this engine permits an incomplete history (i.e. LCP &lt; MSN)
+     * @param minimumSupportedVersion for extended backward compatibility use-cases, the minimum {@link Version} this engine is allowed to read
      */
     public ReadOnlyEngine(
         EngineConfig config,
         SeqNoStats seqNoStats,
         TranslogStats translogStats,
         boolean obtainLock,
         Function<DirectoryReader, DirectoryReader> readerWrapperFunction,
-        boolean requireCompleteHistory
+        boolean requireCompleteHistory,
+        Version minimumSupportedVersion
```

Comment:
I don't think this overloading is required, you can read the `extendedCompatibilitySnapshotVersion` directly from config. `config().getIndexSettings().getExtendedCompatibilitySnapshotVersion()`